src/colorizer/Dataset.ts

```diff
@@ -186,24 +211,53 @@ export default class Dataset {
     return this.frameDimensions || new Vector2(1, 1);
   }
 
-  /** Loads the dataset manifest and features */
-  public async open(): Promise<void> {
+  /** Loads the dataset manifest and features. */
+  public async open(manifestLoader = this.fetchJson): Promise<void> {
     if (this.hasOpened) {
       return;
     }
     this.hasOpened = true;
 
-    const manifest = await this.fetchManifest();
+    const manifest = await manifestLoader(this.manifestUrl);
 
     this.frameFiles = manifest.frames;
     this.featureFiles = manifest.features;
-    this.outlierFile = manifest.outliers;
+
+    // If feature names have units (provided in parentheses at end), strip from the
+    // feature name and save to the units field in the feature metadata unless overrriden.
+    const newFeaturesToFiles: Record<string, string> = {};
+    const featuresToMetadata: Record<string, Partial<FeatureMetaData>> = {};
+    for (const featureName of Object.keys(this.featureFiles)) {
+      const metadata: Partial<FeatureMetaData> = {};
+      let newFeatureName = featureName;
+      // Matches the content inside the first set of parentheses at the end of the string
+      const detectedUnits = featureName.trim().match(/\((.+)\)$/);
+      const metadataUnits = manifest.featureMetadata && manifest.featureMetadata[featureName]?.units;
+
+      if (metadataUnits !== undefined || !detectedUnits) {
+        // Don't change feature name if units are provided in metadata, or if no units
+        // could be found.
+        metadata.units = metadataUnits;
+      } else {
+        // Strip the units from the feature name and save to metadata instead
+        newFeatureName = featureName.replace(detectedUnits[0], "").trim();
+        metadata.units = detectedUnits[1]; // Inner part (inside parentheses)
+      }
+
+      newFeaturesToFiles[newFeatureName] = this.featureFiles[featureName];
+      featuresToMetadata[newFeatureName] = metadata;
```

Comment:
most of the contents of the for loop could be its own function.  I'm not super excited about using these parens as a way to get units in.   I suggest that we disregard parentheses here and don't even try to parse for them, but in the data prep scripts we could do that parsing and populate featureMetadata that way.

Reply:
But since the code is already written, I think just making this code block its own function would be a nice cleanup/improvement.

Reply:
Possible side effect: someone might name their feature with a parenthetical suffix that was not meant to be units.  Python scripts could warn about this...

Reply:
Agree here - finding units in the formatting of name strings could be problematic, but with it already here I'd find it more readable as a separated and named function.

Reply:
Ended up removing this since I made a related change at the data scripts layer! Thank you for the feedback.